### DISCUSSION TITLE
normTop?

### DISCUSSION THREAD

**Discussion by dagarfield** [aertslab/pycisTopic#100] [General]
Hey-o,

Trying to do a parallel analysis to what Ibrahim did in his paper and got to this point in the methods using the R-version of cisTopics

```
We then calculated scores for the topics per region using getRegionsScores with method='NormTop' and scale=TRUE. Finally, we used binarizecisTopics with thrP = 0.975 to get 81 sets of peaks.
```

This functionality isn't exactly paralleled in pycistopic and I was wondering if you could point me to how you'd carry out a similar approach with the python tools. I tried fitting to a gamma distribution, but got pretty ugly results on my own data...before trying again I thought maybe, just maybe, this is already implemented and I missed it. Thanks!

**Comment by SeppeDeWinter** [comment]
Hi @dagarfield 

In pycistopic we have the `binarize_topics` function to binarize topics.

For example:

```python
from pycisTopic.topic_binarization import *
region_bin_topics = binarize_topics(cistopic_obj, method='otsu', ntop=3000, plot=True, num_columns=5, save= outDir + 'topic_binarization/otsu.pdf')
```

Is this what you are looking for?

All the best,

Seppe

**Comment by dagarfield** [comment]
Yes, this function I knew. It was the connection to the previous normTop method that I'm not seeing.
(This function seems to do the job very well, by the way, just for consistency in one of our models of enhancer function)

**Reply by SeppeDeWinter** [reply]
I understand you question now. 

The "NormTop" refers to how the topic distributions are normalised. 
In the pycistopic code the distributions are still normalised, see: https://github.com/aertslab/pycisTopic/blob/8955c76698c7a8a9086b4a58e9fb57f7f7335bf2/pycisTopic/topic_binarization.py#L104C11-L104C11 however the way it's done is not identical indeed. Also the way we binarize the distributions is different now, (otsu vs fitting a gamma distribution).

I guess the easiest way to replicate the R code would be to load your region-topic probabilities in R and using the original cisTopic code. We don't have the code at hand to replicate the R code in python exactly, unfortunately ( maybe @cbravo93 has it somewhere? )

All the best,

Seppe
